pkgs/development/tools/rust/cargo-raze/default.nix

```diff
@@ -19,7 +19,7 @@ rustPlatform.buildRustPackage rec {
   buildInputs = [ curl libgit2 openssl ]
     ++ lib.optional stdenv.isDarwin Security;
 
-  doCheck = true;
+  doCheck = ! stdenv.isDarwin;
```

Comment:
```suggestion
  __darwinAllowLocalNetworking = true;
```

that should fix it.

Reply:
Testing it now.

Reply:
Doesn't work, same issue (with `doCheck = true`), now testing again without `doCheck` since your suggestion removes it, but then I'd be misunderstanding how this change is supposed to fix it.

Reply:
Same issue.

Reply:
``__darwinAllowLocalNetworking = true;`` should allow local socket connections. It does not work all the time but it is worth a try.

Reply:
Oh I see, yes, it's very likely that error is addressed, I just noticed the much more common error: `thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }'`.

Reply:
I will test this change again paying attention to that particular socket failure.

Reply:
`__darwinAllowLocalNetworking = true;` does indeed change the set of tests that fail:

Without (attempt 1):

<details>

```
failures:

---- planning::tests::test_plan_build_produces_aliased_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_aliased_dependencies' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"

Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- planning::tests::test_plan_build_produces_proc_macro_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_proc_macro_dependencies' panicked at 'Cannot build HTTP client: Error { kind: ClientInitialization, context: None, source: Some(Os { code: 24, kind: Other, message: "Too many open files" }), source_type: Some("std::io::error::Error") }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/api/adapter/mod.rs:136:14

---- planning::tests::test_workspace_context_contains_no_binary_dependencies stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_workspace_context_contains_no_binary_dependencies' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_subplan_produces_crate_root_with_forward_slash stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_subplan_produces_crate_root_with_forward_slash' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_workspace_members_share_dependency_of_different_versions stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_workspace_members_share_dependency_of_different_versions' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_semver_matching stdout ----
thread 'planning::tests::test_semver_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpYTCJhH`")', src/planning.rs:600:74

---- settings::tests::test_loading_settings_legacy stdout ----
thread 'settings::tests::test_loading_settings_legacy' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpoMaJSI`")', src/settings.rs:899:91

---- settings::tests::test_loading_workspace_settings stdout ----
thread 'settings::tests::test_loading_workspace_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpfT120m`")', src/settings.rs:932:87

failures:
    planning::tests::test_plan_build_produces_aliased_dependencies
    planning::tests::test_plan_build_produces_proc_macro_dependencies
    planning::tests::test_semver_matching
    planning::tests::test_subplan_produces_crate_root_with_forward_slash
    planning::tests::test_workspace_context_contains_no_binary_dependencies
    planning::tests::test_workspace_members_share_dependency_of_different_versions
    settings::tests::test_loading_settings_legacy
    settings::tests::test_loading_workspace_settings
```

</details>

```
test result: FAILED(B. 48 passed; 8 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.18s
```

Without (attempt 2):

<details>

```
failures:

---- planning::tests::test_binary_dependencies_vendored_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_vendored_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: Too many open files (os error 24)', src/planning.rs:394:8
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- planning::tests::test_semver_matching stdout ----
thread 'planning::tests::test_semver_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', src/testing.rs:147:40

---- planning::tests::test_plan_build_produces_proc_macro_dependencies stdout ----
thread '<unnamed>' panicked at 'error binding to 127.0.0.1:0: error creating server listener: Too many open files (os error 24)', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/hyper/src/server/server.rs:67:17
thread 'planning::tests::test_plan_build_produces_proc_macro_dependencies' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_plan_build_produces_aliased_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_aliased_dependencies' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74

---- planning::tests::test_workspace_context_contains_no_binary_dependencies stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_workspace_context_contains_no_binary_dependencies' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_workspace_members_share_dependency_of_different_versions stdout ----
thread 'planning::tests::test_workspace_members_share_dependency_of_different_versions' panicked at 'Cannot build HTTP client: Error { kind: ClientInitialization, context: None, source: Some(Os { code: 24, kind: Other, message: "Too many open files" }), source_type: Some("std::io::error::Error") }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/api/adapter/mod.rs:136:14

---- planning::tests::test_subplan_produces_crate_root_with_forward_slash stdout ----
thread 'planning::tests::test_subplan_produces_crate_root_with_forward_slash' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74

---- planning::tests::test_binary_dependencies_remote_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_remote_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to download `http://127.0.0.1:62980/api/v1/crates/some-binary-crate/3.3.3/download`

Caused by:
    0: error sending request for url (http://127.0.0.1:62980/api/v1/crates/some-binary-crate/3.3.3/download): error trying to connect: tcp open error: Too many open files (os error 24)
    1: error trying to connect: tcp open error: Too many open files (os error 24)
    2: tcp open error: Too many open files (os error 24)
    3: Too many open files (os error 24)', src/planning.rs:394:8

---- settings::tests::test_loading_package_settings stdout ----
thread 'settings::tests::test_loading_package_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', src/settings.rs:861:54

---- rendering::bazel::tests::additional_build_file_included stdout ----
thread 'rendering::bazel::tests::additional_build_file_included' panicked at 'called `Result::unwrap()` on an `Err` value: Raze failed to render crate "test-library" with cause: "failed to read additional_build_file '/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpdxbv2K/some_additional_build_file': Too many open files (os error 24)"', src/rendering/bazel.rs:654:8

---- planning::tests::test_plan_build_minimum_workspace stdout ----
thread 'planning::tests::test_plan_build_minimum_workspace' panicked at 'called `Result::unwrap()` on an `Err` value: Too many open files (os error 24)', src/metadata.rs:611:54

---- settings::tests::test_loading_settings_legacy stdout ----
thread 'settings::tests::test_loading_settings_legacy' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpsd12GR`")', src/settings.rs:899:91

---- util::tests::detecting_workspace_root stdout ----
thread 'util::tests::detecting_workspace_root' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', src/util.rs:393:61

---- settings::tests::test_loading_workspace_settings stdout ----
thread 'settings::tests::test_loading_workspace_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpiRlFPf`")', src/settings.rs:932:87


failures:
    planning::tests::test_binary_dependencies_remote_genmode
    planning::tests::test_binary_dependencies_vendored_genmode
    planning::tests::test_plan_build_minimum_workspace
    planning::tests::test_plan_build_produces_aliased_dependencies
    planning::tests::test_plan_build_produces_proc_macro_dependencies
    planning::tests::test_semver_matching
    planning::tests::test_subplan_produces_crate_root_with_forward_slash
    planning::tests::test_workspace_context_contains_no_binary_dependencies
    planning::tests::test_workspace_members_share_dependency_of_different_versions
    rendering::bazel::tests::additional_build_file_included
    settings::tests::test_loading_package_settings
    settings::tests::test_loading_settings_legacy
    settings::tests::test_loading_workspace_settings
    util::tests::detecting_workspace_root
```

</details>

```
test result: FAILED(B. 42 passed; 14 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.17s
```


With (attempt 1):

<details>

```
failures:

---- planning::tests::test_workspace_context_contains_no_binary_dependencies stdout ----
thread '<unnamed>' panicked at 'error binding to 127.0.0.1:0: error creating server listener: Too many open files (os error 24)', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/hyper/src/server/server.rs:67:17
thread 'planning::tests::test_workspace_context_contains_no_binary_dependencies' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_plan_build_produces_build_proc_macro_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_build_proc_macro_dependencies' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74

---- planning::tests::test_plan_build_missing_resolve_returns_error stdout ----
thread 'planning::tests::test_plan_build_missing_resolve_returns_error' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/metadata.rs:611:54
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- planning::tests::test_binary_dependencies_remote_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_remote_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to fetch package info from crates.io.

Caused by:
    0: builder error: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:394:8

---- planning::tests::test_workspace_members_share_dependency_of_different_versions stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_workspace_members_share_dependency_of_different_versions' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_subplan_produces_crate_root_with_forward_slash stdout ----
thread '<unnamed>' panicked at 'error binding to 127.0.0.1:0: error creating server listener: Too many open files (os error 24)', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/hyper/src/server/server.rs:67:17
thread 'planning::tests::test_subplan_produces_crate_root_with_forward_slash' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_plan_build_produces_proc_macro_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_proc_macro_dependencies' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74

---- settings::tests::test_loading_settings_legacy stdout ----
thread 'settings::tests::test_loading_settings_legacy' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpVpz7xB`")', src/settings.rs:899:91

---- settings::tests::test_loading_package_settings stdout ----
thread 'settings::tests::test_loading_package_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmp1YpYF1`")', src/settings.rs:863:71

---- planning::tests::test_binary_dependencies_vendored_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_vendored_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: failed to unpack entry at `some-binary-crate-3.3.3/Cargo.toml`

Caused by:
    0: failed to unpack `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpSgyh7x/some-binary-crate-3.3.3/Cargo.toml`
    1: failed to unpack `some-binary-crate-3.3.3/Cargo.toml` into `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpSgyh7x/some-binary-crate-3.3.3/Cargo.toml`
    2: Too many open files (os error 24)', src/planning.rs:394:8

---- settings::tests::test_loading_workspace_settings stdout ----
thread 'settings::tests::test_loading_workspace_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpHdHlLV`")', src/settings.rs:932:87

---- planning::tests::test_semver_matching stdout ----
thread 'planning::tests::test_semver_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to run `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped metadata` with contents:
[package]
name = "test"
version = "0.0.1"

[lib]
path = "not_a_file.rs"


Caused by:
    0: failed to start `cargo metadata`: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:168:74

failures:
    planning::tests::test_binary_dependencies_remote_genmode
    planning::tests::test_binary_dependencies_vendored_genmode
    planning::tests::test_plan_build_missing_resolve_returns_error
    planning::tests::test_plan_build_produces_build_proc_macro_dependencies
    planning::tests::test_plan_build_produces_proc_macro_dependencies
    planning::tests::test_semver_matching
    planning::tests::test_subplan_produces_crate_root_with_forward_slash
    planning::tests::test_workspace_context_contains_no_binary_dependencies
    planning::tests::test_workspace_members_share_dependency_of_different_versions
    settings::tests::test_loading_package_settings
    settings::tests::test_loading_settings_legacy
    settings::tests::test_loading_workspace_settings
```

</details>

```
test result: FAILED(B. 44 passed; 12 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.17s
```

With (attempt 2):

<details>

```
failures:

---- planning::tests::test_workspace_context_contains_no_binary_dependencies stdout ----
thread 'planning::tests::test_workspace_context_contains_no_binary_dependencies' panicked at 'Cannot build HTTP client: Error { kind: ClientInitialization, context: None, source: Some(Os { code: 24, kind: Other, message: "Too many open files" }), source_type: Some("std::io::error::Error") }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/api/adapter/mod.rs:136:14

---- planning::tests::test_workspace_members_share_dependency_of_different_versions stdout ----
thread '<unnamed>' panicked at 'Cannot build local tokio runtime: Os { code: 24, kind: Other, message: "Too many open files" }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1436:14
thread 'planning::tests::test_workspace_members_share_dependency_of_different_versions' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_subplan_produces_crate_root_with_forward_slash stdout ----
thread '<unnamed>' panicked at 'error binding to 127.0.0.1:0: error creating server listener: Too many open files (os error 24)', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/hyper/src/server/server.rs:67:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'planning::tests::test_subplan_produces_crate_root_with_forward_slash' panicked at 'Cannot get server address: RecvError(())', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/lib.rs:1441:37

---- planning::tests::test_plan_build_produces_proc_macro_dependencies stdout ----
thread 'planning::tests::test_plan_build_produces_proc_macro_dependencies' panicked at 'Cannot build HTTP client: Error { kind: ClientInitialization, context: None, source: Some(Os { code: 24, kind: Other, message: "Too many open files" }), source_type: Some("std::io::error::Error") }', /private/tmp/nix-build-cargo-raze-0.12.0.drv-0/cargo-raze-0.12.0-vendor.tar.gz/httpmock/src/api/adapter/mod.rs:136:14

---- planning::tests::test_binary_dependencies_vendored_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_vendored_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to fetch package info from crates.io.

Caused by:
    0: builder error: Too many open files (os error 24)
    1: Too many open files (os error 24)', src/planning.rs:394:8

---- planning::tests::test_binary_dependencies_remote_genmode stdout ----
thread 'planning::tests::test_binary_dependencies_remote_genmode' panicked at 'called `Result::unwrap()` on an `Err` value: Failed to fetch package info from crates.io.

Caused by:
    0: error sending request for url (http://127.0.0.1:62775/api/v1/crates/some-binary-crate): error trying to connect: tcp open error: Too many open files (os error 24)
    1: error trying to connect: tcp open error: Too many open files (os error 24)
    2: tcp open error: Too many open files (os error 24)
    3: Too many open files (os error 24)', src/planning.rs:394:8

---- settings::tests::test_loading_settings_legacy stdout ----
thread 'settings::tests::test_loading_settings_legacy' panicked at 'called `Result::unwrap()` on an `Err` value: Generic("Failed to fetch Metadata with `/nix/store/rgqprchhk58xcsfj7wqcanigbq8h36md-cargo-1.52.1/bin/.cargo-wrapped` from `/private/tmp/nix-build-cargo-raze-0.12.0.drv-0/.tmpqiAahf`")', src/settings.rs:899:91

---- settings::tests::test_loading_workspace_settings stdout ----
thread 'settings::tests::test_loading_workspace_settings' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', src/settings.rs:929:49

---- planning::tests::test_semver_matching stdout ----
thread 'planning::tests::test_semver_matching' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 24, kind: Other, message: "Too many open files" }', src/metadata.rs:562:64


failures:
    planning::tests::test_binary_dependencies_remote_genmode
    planning::tests::test_binary_dependencies_vendored_genmode
    planning::tests::test_plan_build_produces_proc_macro_dependencies
    planning::tests::test_semver_matching
    planning::tests::test_subplan_produces_crate_root_with_forward_slash
    planning::tests::test_workspace_context_contains_no_binary_dependencies
    planning::tests::test_workspace_members_share_dependency_of_different_versions
    settings::tests::test_loading_settings_legacy
    settings::tests::test_loading_workspace_settings
```

</details>

```
test result: FAILED(B. 47 passed; 9 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.26s
```

Reply:
But the variation between runs is high anyway. I can't reproduce any socket related issues locally, but I suspect it might be because I'm not running with the same sandbox settings as hydra.